### DISCUSSION TITLE
Next update?

### DISCUSSION THREAD

**Discussion by weicj** [bangcorrupt/hacktribe#157] [General]
Hi bro, thank you for the costant effort for electribe community. Maybe a little bit wierd but, is there an anticipated roadmap for the next update? like when, and incl what,  XD

**Comment by bangcorrupt** [comment]
Hi, I have no set roadmap or schedule, just work on this when I can.  Next changes will probably be finishing [FX editing](https://github.com/bangcorrupt/hacktribe/wiki/MIDI#fx-edit) and adding groove editing.  Bug #71 needs fixing before going much further, I think that could make motion sequence CC more likely too.  Then maybe some hardware hacks.  There's years worth of ideas in this forum, I'd like to decouple hacktribe from the factory firmware so I can share some source code and people can hack the hack.  Also a bootloader mod to allow easy debricking, and an open source firmware template.  No idea if or when any of this will happen.

**Reply by BKLronin** [reply]
Sounds great to me.

Most excited about the from scratch open source firmware path.

**Reply by maks** [reply]
That makes 2 of us! I'm hopeless at RE and binary patching, but I have done some ARM Dev in the past so I'd be keen to try to hack on a open source firmware. 

The main question for me would be if it's possible to have any open source firmware fairly easy to install without too much chance of bricking? I have somewhere in the cupboard a JTAG debugger but I'd rather not be cracking open my E2 😅

**Reply by weicj** [reply]
Very encouraging! So sad I am so poor at coding, otherwise I would really like to help. 😃

**Reply by testergrooves** [reply]
> Hi, I have no set roadmap or schedule, just work on this when I can. Next changes will probably be finishing [FX editing](https://github.com/bangcorrupt/hacktribe/wiki/MIDI#fx-edit) and adding groove editing. Bug #71 needs fixing before going much further, I think that could make motion sequence CC more likely too. Then maybe some hardware hacks. There's years worth of ideas in this forum, I'd like to decouple hacktribe from the factory firmware so I can share some source code and people can hack the hack. Also a bootloader mod to allow easy debricking, and an open source firmware template. No idea if or when any of this will happen.

I haven't bought an electribe yet but the idea of custom grooves and GUI groove editing makes me desperately want to, even MPC doesn't have that. Hope you add it soon!

**Reply by BKLronin** [reply]
> That makes 2 of us! I'm hopeless at RE and binary patching, but I have done some ARM Dev in the past so I'd be keen to try to hack on a open source firmware.
> 
I only done some medium arduino stuff from scratch and mostly python stuff. DSP and hardware close stuff is very much out of range currently, so I can´t offer ayn help yet. That may change though.

> The main question for me would be if it's possible to have any open source firmware fairly easy to install without too much chance of bricking? I have somewhere in the cupboard a JTAG debugger but I'd rather not be cracking open my E2 sweat_smile

I at least have a stm32(?) progammer somewhere around.

**Reply by bangcorrupt** [reply]
> custom grooves

This is already partly possible if you're comfortable with Python.  See [e2_groove_template.py](https://github.com/bangcorrupt/hacktribe/blob/main/scripts/e2_groove_template.py) for an example.  Groove templates can't be saved to the flash yet, so we have to export them after editing.  In the future `E2Groove` class will be changed  to use the struct definitions in [e2_formats.py](https://github.com/bangcorrupt/hacktribe/blob/main/app/e2_formats.py), so I wouldn't build anything major on top of it.  In time I'll add NRPN control of groove templates, then (hopefully) editing using the controls on the device.

> install without too much chance of bricking

I started work on a bootloader hack for recovery from a bad firmware update.  Firmware update functions should never be part of the updated firmware.

> JTAG debugger

A JLink is good for the CPU.  A Raspberry Pi will also work, but it's a bit slower.
